nesta/production/routines/arxiv/deepchange_analysis_task.py

```diff
@@ -0,0 +1,270 @@
+        with open(DEEPCHANGE_QUERY) as sql_query:
+            df = pd.read_sql(sql_query.read(), self.engine)
+        logging.info(f"Retrieved {len(df)} articles from database")
+
+        # collect topics, determine which represents deep_learning and apply flag
+        dl_topic_ids = dc.get_article_ids_by_term(self.engine,
+                                                  term='deep_learning',
+                                                  min_weight=0.2)
+        df['is_dl'] = df.article_id.apply(lambda i: i in dl_topic_ids)
+        logging.info(f"Flagged {df.is_dl.sum()} deep learning articles")
+
+        df['date'] = df.apply(lambda row: row.article_updated or row.article_created,
+                              axis=1)
+        df['year'] = df.date.apply(lambda date: date.year)
+        df = dc.add_before_date_flag(df,
+                                     date_column='date',
+                                     before_year=YEAR_THRESHOLD)
+
+        # TODO: decide if we are applying de-duplication for each chart
+        deduped = df.drop_duplicates('article_id')
+
+        # first plot - dl/non dl distribution by country (top 20)
+        pivot_by_country = (pd.pivot_table(deduped.groupby(['institute_country', 'is_dl'])
+                                           .size()
+                                           .reset_index(drop=False),
+                            index='institute_country',
+                            columns='is_dl',
+                            values=0)
+                            .apply(lambda x: 100 * (x / x.sum())))
+
+        fig, ax = plt.subplots(figsize=(7, 4))
+        (pivot_by_country.sort_values(True, ascending=False)[:20]
+                         .sort_values(True)
+                         .plot.barh(ax=ax))
+        ax.set_xlabel('Percentage of DL papers in country')
+        ax.set_ylabel('%')
+        dc.plot_to_s3('arxlive-charts', 'figure_1.png', plt)
+
+        # second plot - dl/non dl distribution by city (top 20)
+        pivot_by_city = (pd.pivot_table(deduped.groupby(['institute_city', 'is_dl'])
+                                        .size()
+                                        .reset_index(drop=False),
+                         index='institute_city',
+                         columns='is_dl',
+                         values=0)
+                         .apply(lambda x: 100 * (x / x.sum())))
+
+        fig, ax = plt.subplots(figsize=(7, 4))
+        (pivot_by_city.sort_values(True, ascending=False)[:20]
+                      .sort_values(True)
+                      .plot.barh(ax=ax))
+        ax.set_xlabel('Percentage of DL papers in city')
+        ax.set_ylabel('%')
+        dc.plot_to_s3('arxlive-charts', 'figure_2.png', plt)
+
+        # third plot - percentage of dl papers by year
+        # TODO: set reasonable limits on the y axis
+        papers_by_year = pd.crosstab(df['year'], df['is_dl']).loc[2000:]
```

Comment:
1) I would add `2000` as a parameter
2) What does this number correspond to?

Reply:
added a comment to explain, as discussed